hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/domain/ContractResultServiceImpl.java

```diff
@@ -88,15 +88,18 @@ public void process(@NonNull RecordItem recordItem, Transaction transaction) {
 
         // contractResult
         var transactionHandler = transactionHandlerFactory.get(TransactionType.of(transaction.getType()));
-
+        var throwRecoverableError = !(recordItem.isSuccessful()
+                && (transactionRecord.getReceipt().getContractID().hasEvmAddress()));
         // in pre-compile case transaction is not a contract type and entityId will be of a different type
         var contractId = (contractCallOrCreate
                         ? Optional.ofNullable(transaction.getEntityId())
-                        : entityIdService.lookup(functionResult.getContractID()))
+                        : entityIdService.lookup(functionResult.getContractID(), throwRecoverableError))
                 .orElse(EntityId.EMPTY);
-        var isRecoverableError = EntityId.isEmpty(contractId)
-                && !contractCallOrCreate
-                && !ContractID.getDefaultInstance().equals(functionResult.getContractID());
+        var isRecoverableError =
+                !(recordItem.isSuccessful() && functionResult.getContractID().hasEvmAddress())
```

Comment:
I might've misled you before. Believe `functionResult.getContractID()` should already log above if it had a recoverable error. This check should run for the one in receipt.

Also, we want to ensure we don't log twice like we're doing now on main so we can skip if previously logged. Something like this (please verify):

```suggestion
                !throwRecoverableError &&
                !(recordItem.isSuccessful() && transactionRecord.getReceipt().getContractID().hasEvmAddress())
```

Reply:
Thanks. I ran through the recreate scenario and updated this section. 

The first boolean, `throwRecoverableError` wouldn't indicate that an error was thrown, only that it could be thrown. I've changed the name and added a comment.